### DISCUSSION TITLE
AutoFixture.ObjectCreationExceptionWithPath ->it occurs but dont know why..

### DISCUSSION THREAD

**Discussion by endend789** [AutoFixture/AutoFixture#1405] [Q&A]
I downloaded moq, autofixture, and autofixture.moq to the nuget package and generated test case through Unitverse. And then this message occured.
I asked Unitverse about it and they(Unitverse) said it was not their error, but sound like an issue in autofixture. The test won't proceed due to this error, do you know why?
![image](https://github.com/AutoFixture/AutoFixture/assets/114501864/c94db633-c2e9-4eb3-9d1a-0e179506125a)

**Comment by aivascu** [comment]
@endend789 this is likely a configuration issue. Could you provide a reproducible code sample?

**Reply by endend789** [reply]
Original code is confidential, so i made a sample code and the luckily the error message is 100% same.
(sample one is simple.)
**<Sample code>**

```cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sample
{
    public class Calculation
    {
        static void Main() { }
        public int FindMax(int[] arr)
        {
            int max = 0;
            for (int i = 0; i < arr.Length; i++)
            {
                if (max < arr[i])
                    max = arr[i];
            }
            return max;
        }
        public int FindMin(int[] arr)
        {
            int min = 0;
            for (int i = 0; i < arr.Length; i++)
            {
                if (min > arr[i])
                    min = arr[i];
            }
            return min;
        }
    }
}
```

**<Sample's Test code>**
(used moq, autofixture, autofixture.moq and generated with Unitverse)
```cs
namespace example.UnitTest
{
    using System;
    using AutoFixture;
    using AutoFixture.AutoMoq;
    using example;
    using Microsoft.VisualStudio.TestTools.UnitTesting;
    using Sample;

    [TestClass]
    public class exampleTest
    {
        [TestMethod]
        public void TestMethod1()
        {
        }
    }

    [TestClass]
    public class CalculationTest
    {
        private Calculation _testClass;
        private IFixture _fixture;

        [TestInitialize]
        public void SetUp()
        {
            _fixture = new Fixture().Customize(new AutoMoqCustomization());
            _testClass = _fixture.Create<Calculation>(); // <= **Error occurs here!!**
        }

        [TestMethod]
        public void FindMax_MethodBasicTest()
        {
            // Arrange
            var arr = _fixture.Create<int[]>();

            // Act
            var result = _testClass.FindMax(arr);

            // Assert
            Assert.Fail("Create or modify test");
        }

        [TestMethod]
        public void FindMax_MethodArrWithNullTestN()
        {
            Assert.ThrowsException<ArgumentNullException>(() => _testClass.FindMax(default(int[])));
        }

        [TestMethod]
        public void FindMin_MethodBasicTest()
        {
            // Arrange
            var arr = _fixture.Create<int[]>();

            // Act
            var result = _testClass.FindMin(arr);

            // Assert
            Assert.Fail("Create or modify test");
        }

        [TestMethod]
        public void FindMin_MethodArrWithNullTestN()
        {
            Assert.ThrowsException<ArgumentNullException>(() => _testClass.FindMin(default(int[])));
        }
    }
}
```

<Error message(screen shot)>
![image](https://github.com/AutoFixture/AutoFixture/assets/114501864/3734f012-06d7-467b-9c53-6d47cee6fabc)
(same with original one)

**<Original code(Showing error point)>**
```cs
[TestInitialize]
        public void SetUp()
        {
            _fixture = new Fixture().Customize(new AutoMoqCustomization());
            _accessMode = _fixture.Create<AccessMode>(); <= Error!
            _authority = _fixture.Create<Authority>();
            _sSettingsFilePath = _fixture.Create<string>();
            _sLogFolderPath = _fixture.Create<string>();
            _testClass = _fixture.Create<EmbeddedBrowserViewModel>();
        }
```

**Reply by aivascu** [reply]
@endend789 I tried your code sample and no error was thrown. Are you sure your example reproduces the error?
Could you check the inner exception as the exception text suggests? Maybe post it here if it's possible.

Here's my sample:
I ran this sample on .NET 4.8 an .NET 7
<details>
<summary>Tested class</summary>

```cs
public class Calculation
{
    static void Main() { }

    public int FindMax(int[] arr)
    {
        int max = 0;
        for (int i = 0; i < arr.Length; i++)
        {
            if (max < arr[i])
                max = arr[i];
        }
        return max;
    }

    public int FindMin(int[] arr)
    {
        int min = 0;
        for (int i = 0; i < arr.Length; i++)
        {
            if (min > arr[i])
                min = arr[i];
        }
        return min;
    }
}
```
</details>

```cs
[TestClass]
public class CalculationTest
{
    private Calculation _testClass;
    private IFixture _fixture;

    [TestInitialize]
    public void SetUp()
    {
        _fixture = new Fixture().Customize(new AutoMoqCustomization());
        _testClass = _fixture.Create<Calculation>(); // <= No exception here
    }

    [TestMethod]
    public void FindMax_MethodBasicTest()
    {
        // Arrange
        var arr = _fixture.Create<int[]>();

        // Act
        var result = _testClass.FindMax(arr);

        // Assert
        Assert.AreEqual(arr.Max(), result);
    }

    [TestMethod]
    public void FindMax_MethodArrWithNullTestN()
    {
        Assert.ThrowsException<NullReferenceException>(() => _testClass.FindMax(default!));
    }

    [TestMethod]
    public void FindMin_MethodBasicTest()
    {
        // Arrange
        var arr = _fixture.Create<int[]>();

        // Act
        var result = _testClass.FindMin(arr);

        // Assert
        Assert.AreEqual(arr.Min(), result); // This fails cause the min calculation is bugged.
    }

    [TestMethod]
    public void FindMin_MethodArrWithNullTestN()
    {
        Assert.ThrowsException<NullReferenceException>(() => _testClass.FindMin(default!));
    }
}
```

**Reply by endend789** [reply]
here's inner exception message(if its weird sentence, sorry i used translator)

Inner exception message:
System.IO.FileLoadException: Could not load file or assembly 'Moq, Version=4.1.1308.2120, Culture=neutral, PublicKeyToken=69f491c39445e920' or one of the files or assemblies listed here. The assembly reference does not match the manifest definition of the located assembly. (Exception is HRESULT: 0x80131040)

today i changed some things' version and it worked. before(when error occured), i used
visual studio: 2022_17_5_1
.NET: 4.7.2
moq: 4.20.69
autofixture: 4.18.0
autofixture.automoq: 4.18.0

i modified version(no error).
.NET: 4.6.2
moq: 4.1.1308.2120

on the inner exception message, there is a 'Moq, Version=4.1.~~'. So i changed it and it worked.
i cant change visual studio version, but can change .NET, moq, autofixture, auto~.automoq.
can you tell me their versions you used?

**Reply by aivascu** [reply]
@endend789 the inner exception mentions that the test runner couldn't find the Moq package. AutoFixture is fully compatible with the latest version of Moq so there shouldn't be any issues there.
Make sure you have installed the latest version of Moq in your test projects. If you're using the old project file (*.csproj) format reinstall the packages and make sure that the package.config, app.config and the *.csproj mention the same package version and target framework. Also check that the assembly redirects are configured correctly.
Let me know if that helped.
